test/Numerics/DGMethods/advection_diffusion/fvm_spherical_hyperdiffusion.jl

```diff
@@ -0,0 +1,180 @@
+using MPI
+using ClimateMachine
+using Logging
+using ClimateMachine.Mesh.Topologies
+using ClimateMachine.Mesh.Grids
+using ClimateMachine.DGMethods
+using ClimateMachine.DGMethods.NumericalFluxes
+import ClimateMachine.DGMethods.FVReconstructions: FVConstant, FVLinear
+
+using ClimateMachine.MPIStateArrays
+using LinearAlgebra
+using Printf
+using Dates
+using ClimateMachine.GenericCallbacks:
+    EveryXWallTimeSeconds, EveryXSimulationSteps
+using ClimateMachine.ODESolvers
+using ClimateMachine.VTK: writevtk, writepvtu
+using ClimateMachine.Mesh.Grids: min_node_distance
+using ClimateMachine.Atmos: SphericalOrientation, latitude, longitude
+
+using ClimateMachine.Orientations
+using CLIMAParameters
+using CLIMAParameters.Planet
+
+struct EarthParameterSet <: AbstractEarthParameterSet end
+const param_set = EarthParameterSet()
+CLIMAParameters.Planet.planet_radius(::EarthParameterSet) = 60e3
+
+const output = parse(Bool, lowercase(get(ENV, "JULIA_CLIMA_OUTPUT", "false")))
+
+if !@isdefined integration_testing
+    const integration_testing = parse(
+        Bool,
+        lowercase(get(ENV, "JULIA_CLIMA_INTEGRATION_TESTING", "false")),
+    )
+end
+
+include("hyperdiffusion_model.jl")
+
+struct ConstantHyperDiffusion{dim, dir, FT} <: HyperDiffusionProblem
+    D::FT
+    l::FT
+    m::FT
+end
+
+function nodal_init_state_auxiliary!(
+    balance_law::HyperDiffusion,
+    aux::Vars,
+    tmp::Vars,
+    geom::LocalGeometry,
+)
+    FT = eltype(aux)
+    aux.D = balance_law.problem.D * SMatrix{3, 3, Float64}(I)
+    aux.coord = geom.coord
+end
+
+"""
+    initial condition is given by ρ0 = ρ0(r)
+    test: ∇^4_horz ρ0(r) = 0
+"""
+
+function initial_condition!(
+    problem::ConstantHyperDiffusion{dim, dir},
+    state,
+    aux,
+    x,
+    t,
+) where {dim, dir}
+    @inbounds begin
+        FT = eltype(state)
+        # import planet paraset
+        _a::FT = planet_radius(param_set)
+        r = norm(aux.coord)
+        z = r - _a
+        state.ρ = cos(z / 30.0e3)
```

Comment:
Is there a way to access domain height in initial_condition! It's better to write `cos(z/height)` here